### ISSUE TITLE
Logic for PullRequest.needs_reposting seems wrong

### ISSUE THREAD

**Issue by monfresh** [thoughtbot/pester#138]
Perhaps I'm misunderstanding how this is supposed to work, but it seems to me that the only thing that should determine whether a pull request needs to be reposted is if its status is "needs review". Currently, the logic also checks to see if `reposted_at` is `nil` and when the PR was last updated.

If I understand the code correctly, that means that once a PR is reposted, it will never be reposted again, even if it still needs to be reviewed.

As far as I can tell, in order for the `repost_prs` rake task to run, it requires you to create a new job that runs every x minutes on Heroku via the scheduler add-on. You have to manually define the frequency, so I'm not sure what value the `REPOST_THRESHOLD` env var provides since it doesn't map to the job frequency.

The way I envision this should work is that the job runs every x minutes, and checks to see which pull requests need to be posted to Slack, based solely on their status. As long as a PR still needs to be reviewed, then it should be posted to Slack, regardless of whether it was previously reposted, or when it was last updated.

Am I misunderstanding anything?

**Comment by drapergeek** [comment]
>If I understand the code correctly, that means that once a PR is reposted, it will never be reposted again, even if it still needs to be reviewed.

This is correct, we only repost PRs one time. If a PR has been posted twice over a period of time and still hasn't gotten a review, it won't get one with another message to the room. We didn't want our slack to be filled with automated messages. Our team is pretty sensitive to noise in slack channels so we keep messages that aren't necessary to a minimum.

>As far as I can tell, in order for the repost_prs rake task to run, it requires you to create a new job that runs every x minutes on Heroku via the scheduler add-on. You have to manually define the frequency, so I'm not sure what value the REPOST_THRESHOLD env var provides since it doesn't map to the job frequency.

The `REPOST_THRESHOLD` is how long we should be waiting before we repost the PR. We wanted to make this configurable. 30 minutes seems appropriate for our situation.

>The way I envision this should work is that the job runs every x minutes, and checks to see which pull requests need to be posted to Slack, based solely on their status. As long as a PR still needs to be reviewed, then it should be posted to Slack, regardless of whether it was previously reposted, or when it was last updated.

That works a lot of the times but there are cases where it turns out to be less ideal. Let's say that we set the job to run every 30 minutes. If we put in a PR at 11:29, when the job runs at 11:30, it's going to repost that again because no one has signed off. So we now have two automated messages when we should have had one. The combination of a repost threshold and running the job often (we have ours at 10 minutes) means that each post gets reposted within a reasonable time without completely filling up the channel with messages.

**Comment by monfresh** [comment]
Thanks for the details. Those are great points. I obviously haven't played with Pester long enough. I totally get the point about noise sensitivity. One thought I had was to add an additional check so that if a PR still hasn't been reviewed after 24 hours, then a new message gets posted to Slack, and then keeps posting once a day while the PR still needs review. 

On one of our projects with limited dedicated resources, we often end up with PRs that wait for reviews for days, sometimes weeks. We'd like to fix that, so we thought we'd give Pester a try.

What do you think about adding the option to configure a second pester with a separate threshold?

**Comment by drapergeek** [comment]
>What do you think about adding the option to configure a second pester with a separate threshold?

I'm definitely open to new ways of notifying people so definitely give it a shot. Basically rules are:
* Does it keep the code simple
* Is is configurable

As long as that applies, I'm happy to incorporate it
